README.md

```diff
@@ -5,42 +5,129 @@
  <p align="center">Helper library to sign a transaction
 </p>
 
-## Install
+The prefix `\x19LSP6 ExecuteRelayCall:\n` is added to prevent users from signing a transaction while thinking it's a message.
+
+- [GitHub Repository](https://github.com/lukso-network/tools-lsp6-signer)
+- [NPM Package](TODO)
+
+# Install
 
 ```bash
 npm install @lukso/lsp6-signer.js
 ```
 
-## Setup
+# Setup
 
 ```javascript
-import { LSPFactory } from '@lukso/lsp6-signer.js';
+import { LSP6Signer } from '@lukso/lsp6-signer.js';
 
 const lsp6Signer = new LSP6Signer();
 ```
 
-## Usage
+# Usage
 
-### Hash a message
+## hashMessage
 
 ```javascript
-const hash = lsp6Signer.hashMessage(message);
+lsp6Signer.hashMessage(message);
 ```
 
-### Sign a transaction
+Hashes the given message. The data will be enveloped as follows: `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message` and hashed using keccak256.
+
+### Parameters
+
+`message` - `String`: A message to hash
+
+### Returns
+
+`String`: The hashed message
+
+### Example
 
 ```javascript
-const signedObject = lsp6Signer.sign(message, signingKey);
+lsp6Signer.hash('Hello World') >
+  '0x90379a0a840499a5e18db4ff7b4d9a56cd009f014a66e3ca86001da3442b7d2c';
 ```
 
-### Recover an adress used to sign a transaction
+## sign
+
+```javascript
+lsp6Signer.sign(message, signingKey);
+```
+
+Signs the message.
+
+### Parameters
+
+`message` - `String`: The message to sign. \
+`signingKey` - `String`: The private key to sign with.
+
+> **_NOTE:_** The value passed as the data parameter will be wrapped as follows: `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`.
+
+### Returns
+
+`Object`: **The Message object**
+
+- `message` - `String`: The given message.
+- `messageHash` - `String`: The hash of the given message.
+- `r` - `String`: First 32 bytes of the signature.
+- `s` - `String`: Next 32 bytes of the signature.
+- `v` - `String`: Recovery value + 27.
+
+### Example
+
+```javascript
+lsp6Signer.sign(
+  'hello',
+  'ffeb17b9a6059fec3bbab63d76b060b7380cac7a62ce6621a134531a46458968',
+) >
+  {
+    message: 'hello',
+    messageHash:
+      '0xd8fd712232755d5f8819c95fdd911dcd51c012b8a66f9710ef0c86ad30925fa6',
+    v: '0x1c',
+    r: '0x1eab2de0103b8e82650f9706b17cf2adce55a335e7041bad5a94ab49c56a9c12',
+    s: '0x662e80a369ffa2a6a77fbeaad1f32653cbd74860c8fbc999b1fc47b8d1cb7d93',
+    signature:
+      '0x1eab2de0103b8e82650f9706b17cf2adce55a335e7041bad5a94ab49c56a9c12662e80a369ffa2a6a77fbeaad1f32653cbd74860c8fbc999b1fc47b8d1cb7d931c',
+  };
+```
+
+## recover
+
+```javascript
+lsp6Signer.recover(message, signature, isMessagePrefixed);
+```
+
+Recovers the LUKSO address which was used to sign the given data.
```

Comment:
The address is not specific to LUKSO. The private key is mathematically linked to the public address so you could use the same private key on ethereum, LUKSO, Binance smart chain etc etc 

Also we want to accommodate  people using these tools on any chain they like, so we can keep these docs generic about the network they are used on